reana_server/gitlab_client.py

```diff
@@ -0,0 +1,266 @@
+# This file is part of REANA.
+# Copyright (C) 2024 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""REANA-Server GitLab client."""
+
+from typing import Dict, Optional, Union
+from urllib.parse import quote_plus
+import requests
+import yaml
+
+from reana_commons.k8s.secrets import REANAUserSecretsStore
+
+from reana_server.config import REANA_GITLAB_HOST
+
+
+class GitLabException(Exception):
```

Comment:
Cosmetics: since GitLabClient is our own, perhaps we can name all exceptions starting with this prefix, to better associate them together and make sure that the exceptions are not "generic" or coming from some other library: 

- GitLabClientException
- GitLabClientRequestError
- GitLabClientInvalidToken
- ...